ClosedXML/Excel/CalcEngine/Functions/Text.cs

```diff
@@ -293,6 +294,19 @@ private static object Value(List<Expression> p)
             return double.Parse((string)p[0], NumberStyles.Any, CultureInfo.InvariantCulture);
         }
 
+        private static object NumberValue(List<Expression> p)
```

Comment:
Add some error checking and throw the appropriate exceptions. I see you already referenced the `ClosedXML.Excel.CalcEngine.Exceptions` namespace at the top of the file.